src/fsharp/ReferenceLoading.PaketHandler.fs

```diff
@@ -0,0 +1,232 @@
+                else
+                    try File.Delete(loadScript) with _ -> ()
+                    let toolPath = alterToolPath toolPath
+                    File.WriteAllLines(workingDirSpecFile.FullName, packageManagerTextLines)
+                    printfn "running package resolution in '%s'..." workingDir
+                    let startInfo = 
+                        System.Diagnostics.ProcessStartInfo(
+                            FileName = toolPath,
+                            WorkingDirectory = workingDir, 
+                            RedirectStandardOutput = true,
+                            RedirectStandardError = true,
+                            Arguments = getCommand targetFramework rootDir,
+                            CreateNoWindow = true,
+                            UseShellExecute = false)
+                
+                    use p = new System.Diagnostics.Process()
+                    let errors = System.Collections.Generic.List<_>()
+                    let log = System.Collections.Generic.List<_>()
+                    p.StartInfo <- startInfo
+                    p.ErrorDataReceived.Add(fun d -> if d.Data <> null then errors.Add d.Data)
+                    p.OutputDataReceived.Add(fun d -> if d.Data <> null then log.Add d.Data)
+                    p.Start() |> ignore
+                    p.BeginErrorReadLine()
+                    p.BeginOutputReadLine()
+                    p.WaitForExit()
+
+                    printfn "done running package resolution..."
+                    if p.ExitCode <> 0 then
+                        let msg = String.Join(Environment.NewLine, errors)
+                        PackageResolutionFailed(toolPath, workingDir, msg)
+                    else
+                        printfn "package resolution completed at %A" System.DateTimeOffset.UtcNow
+                        Solved(loadScript,additionalIncludeFolders())
+
+    let getLoadScript baseDir packageManagerLoadScriptSubDirectory loadScriptName =
+      System.IO.Path.Combine(baseDir, packageManagerLoadScriptSubDirectory, loadScriptName)
+
+    /// Resolves absolute load script location: something like
+    /// baseDir/.paket/load/scriptName
+    /// or
+    /// baseDir/.paket/load/frameworkDir/scriptName 
+    let GetPaketLoadScriptLocation baseDir optionalFrameworkDir scriptName =
+      let paketLoadFolder = System.IO.Path.Combine(Internals.PM_DIR,"load")
+      getLoadScript
+        baseDir
+        (match optionalFrameworkDir with | None -> paketLoadFolder | Some frameworkDir -> System.IO.Path.Combine(paketLoadFolder, frameworkDir))
+        scriptName
+
+
+module GenericHandler =
+    type IPragmaProvider =
+        inherit System.IDisposable
+        abstract PragmaName : string
+        abstract ProcessArg : string -> unit
+        abstract HasPendingChanges : bool
+        abstract ChangesToScript : string -> (string * string list) option
+
+    type PaketPragmaHandler() =
```

Comment:
Yes, in the one for F# compiler (which is not src/fsharp/ReferenceLoading.PaketHandler.fs but the other one I added with few small members).